### PR TITLE
fix: #693 Add composite way

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

File: mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/dayReportDTOToDayReportEmpty.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import {SchemasDayReportPopulatedResponse} from "src/apiAutogenerated";
+import {DayReport} from "src/model/businessModel/DayReport";
+
+/**
+ * Convert {@link DayReportDTO} to {@link DayReport}
+ */
+export const DayReportDTOToDayReportEmpty = (dayReportDTO: SchemasDayReportPopulatedResponse): DayReport => {
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/dayReportDTOToDayReportEmpty.ts:7]: optimise

File: mw-webapp/src/utils/createCompositeWay.ts [hunk 2]
```diff
@@ -26,10 +26,10 @@ export const createCompositeWay = (way: Way): Way => {
...
     const sortedDayReport = new DayReport({
       ...updatedDayReport,
-      jobsDone: updatedDayReport.jobsDone.sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
-      plans: updatedDayReport.plans.sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
-      problems: updatedDayReport.problems.sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
-      comments: updatedDayReport.comments.sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
+      jobsDone: updatedDayReport.jobsDone.slice().sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
+      plans: updatedDayReport.plans.slice().sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
+      problems: updatedDayReport.problems.slice().sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
+      comments: updatedDayReport.comments.slice().sort((a, b) => DateUtils.datesASCSorter(a.createdAt, b.createdAt)),
```
Comment (tritonJS826) [inline mw-webapp/src/utils/createCompositeWay.ts:32]: why do we need slice here?

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/utils/createCompositeWay.ts (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/createCompositeWay.ts:32]: MobX doesn't work without slice

File: mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/dayReportDTOToDayReportEmpty.ts (hunk 1 above)
Comment (Ekaterina1994) [reply mw-webapp/src/dataAccessLogic/DTOToPreviewConverter/dayReportDTOToDayReportEmpty.ts]: Done

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/utils/createCompositeWay.ts (hunk 2 above)
Comment (Ekaterina1994) [reply mw-webapp/src/utils/createCompositeWay.ts:33]: Created issue #996

**Pull request merged**